notebooks/problem-sets/quantum_enigma_problem_set.ipynb

```diff
@@ -0,0 +1,298 @@
+    "</details>\n",
+    "<!-- ::: -->\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "goals": [
+     {
+      "id": "enigma-quiz-1"
+     }
+    ],
+    "grader_answer": "qc",
+    "grader_id": "quantum_enigmas/problem1"
+   },
+   "outputs": [],
+   "source": [
+    "from qiskit import QuantumCircuit, Aer, transpile\n",
+    "from qiskit.visualization import plot_histogram\n",
+    "\n",
+    "# Start your work here.\n",
+    "# We've provided the circuit that is shown above\n",
+    "# Your circuit MUST be named qc\n",
+    "\n",
+    "qc = QuantumCircuit(3)\n",
+    "\n",
+    "qc.h(0)\n",
+    "qc.h(2)\n",
+    "qc.cx(0, 1)\n",
+    "qc.barrier(0, 1, 2)\n",
+    "qc.cx(2, 1)\n",
+    "qc.x(2)\n",
+    "qc.cx(2, 1)\n",
+    "qc.x(2)\n",
+    "qc.barrier(0, 1, 2)\n",
+    "qc.swap(0, 1)\n",
+    "qc.x(1)\n",
+    "qc.cx(2, 1)\n",
+    "qc.x(0)\n",
+    "qc.x(2)\n",
+    "qc.cx(2, 1)\n",
+    "qc.x(2)\n",
+    "\n",
+    "# Execute the circuit and draw the histogram\n",
+    "\n",
+    "measured_qc = qc.measure_all(inplace=False)\n",
+    "backend = Aer.get_backend('qasm_simulator') # the device to run on\n",
+    "result = backend.run(transpile(measured_qc, backend), shots=1024).result()\n",
+    "counts  = result.get_counts(measured_qc)\n",
+    "plot_histogram(counts)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem 2\n",
+    "\n",
+    "<!-- ::: q-block.exercise -->\n",
+    "\n",
+    "### Quick quiz\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"enigma-quiz-2\") -->\n",
+    "\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "Can you interpret the results of Question 1?\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "1. After simplification, $q_0$, $q_1$, and $q_2$ remain entangled altogether.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "2. After simplification, q0 and q1 are entangled with a H and a CNOT gates, while q2 only has a H gate.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option(correct) -->\n",
+    "\n",
+    "3. After simplification, we finally know which guardian is lying.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem 3\n",
+    "\n",
+    "<!-- ::: q-block.exercise -->\n",
+    "\n",
+    "### Quick quiz\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"enigma-quiz-3\") -->\n",
+    "\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "Launching algorithm on modern quantum computers do not always lead to 100% successful results as some noise bring a portion of bad results. If you launch the whole circuit on a real quantum computer, what is the percentage of good answers you get?\n",
```

Comment:
Should we provide the code again and a backend to use?

Reply:
We _can_ but I'm concerned the real backend will take an hour to run, given the queue

Should we add a disclaimer that a student won't necessary _need_ to run it to see the answer?

Reply:
[Here's an example](https://learn.qiskit.org/course/introduction/why-quantum-computing) of how it would look

Reply:
I like it with the disclaimer.

Reply:
Sounds great, will add that shortly!

Reply:
Added to Problem 3